lib/topics_filter.rb

```diff
@@ -370,19 +370,26 @@ def tag_ids_from_tag_names(tag_names)
     all_tag_ids
   end
 
-  def tag_groups_to_tags(values)
-    values.map do |key_prefix, value|
-      tag_groups_names =
-        if value.include?(",")
-          value.split(",")
-        else
-          [value]
-        end
-      tags = TagGroup.where(name: tag_groups_names).map(&:tags).flatten.map(&:name).join(",")
-      [key_prefix, tags]
+  def filter_tag_groups(values:)
+    values.each do |key_prefix, tag_groups|
+      should_exclude = "NOT" if key_prefix == "-"
+      tag_group_ids = TagGroup.where(name: tag_groups).pluck(:id)
```

Comment:
You should make sure here the current user can see the loaded tag groups here. I think you can use the `visible` method in the TagGroup model for that.